## Week 13: Generate an RNA-Seq count matrix

We will be using the dataset from the Biostar Workflows: [RNA-Seq with Hisat2 tutorial](https://www.biostarhandbook.com/books/workflows/rnaseq/rnaseq-using-hisat/).

### Instructions:

Make sure to have the Biostar Toolbox in the same directory
```
# Get the toolbox code
bio code
```

Step 1. Create a design.csv file that contains the following:
```
sample,group
HBR_1,HBR
HBR_2,HBR
HBR_3,HBR
UHR_1,UHR
UHR_2,UHR
UHR_3,UHR
```

Step 2. Create count.txt

Activate bioinfo environment
```
conda activate bioinfo
```

Run the count pipeline
```
make count
```

Step 3. Do post-count process

Activate stats environment
```
conda activate stats
```

Run the post-count pipeline
```
make post
```

We should get a counts.csv file as shown below:

![](images/counts.png)

First we see that the data is sparse (a lot of zeros). We check some gene entries that can give us insight.

![](images/compare.png)

Here, we can see by eye-balling that the counts of HBR's and UHR's are clearly delineated.

Next, we verify in IGV that indeed what we have are RNA's

![](images/igv.png)

Looking at the red arches, indicating that the splice-aware alignment by Hisat2 is working and at the bottom clearly showing exons. These are indeed RNA's.
